connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -255,12 +257,29 @@ public Response putConnectorConfig(final @PathParam("connector") String connecto
 
     @POST
     @Path("/{connector}/restart")
-    public void restartConnector(final @PathParam("connector") String connector,
+    public Response restartConnector(final @PathParam("connector") String connector,
                                  final @Context HttpHeaders headers,
+                                 final @DefaultValue ("false") @QueryParam("includeTasks") Boolean includeTasks,
+                                 final @DefaultValue ("false") @QueryParam("onlyFailed") Boolean onlyFailed,
```

Comment:
Nice! I didn't recall that `@DefaultValue` exists. But we should not have a space between that annotation and the open parenthesis:
```suggestion
                                 final @DefaultValue("false") @QueryParam("includeTasks") Boolean includeTasks,
                                 final @DefaultValue("false") @QueryParam("onlyFailed") Boolean onlyFailed,
```